pumpkin-core/src/random/mod.rs

```diff
@@ -1,13 +1,139 @@
+        }
+    }
+
+    pub fn next_i32(&mut self) -> i32 {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_i32(),
+            Self::Legacy(rand) => rand.next_i32(),
+        }
+    }
+
+    pub fn next_bounded_i32(&mut self, bound: i32) -> i32 {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_bounded_i32(bound),
+            Self::Legacy(rand) => rand.next_bounded_i32(bound),
+        }
+    }
+
+    pub fn next_inbetween_i32(&mut self, min: i32, max: i32) -> i32 {
+        self.next_bounded_i32(max - min + 1) + min
+    }
+
+    pub fn next_i64(&mut self) -> i64 {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_i64(),
+            Self::Legacy(rand) => rand.next_i64(),
+        }
+    }
+
+    pub fn next_bool(&mut self) -> bool {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_bool(),
+            Self::Legacy(rand) => rand.next_bool(),
+        }
+    }
+
+    pub fn next_f32(&mut self) -> f32 {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_f32(),
+            Self::Legacy(rand) => rand.next_f32(),
+        }
+    }
+
+    pub fn next_f64(&mut self) -> f64 {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_f64(),
+            Self::Legacy(rand) => rand.next_f64(),
+        }
+    }
+
+    pub fn next_gaussian(&mut self) -> f64 {
+        match self {
+            Self::Xoroshiro(rand) => rand.next_gaussian(),
+            Self::Legacy(rand) => rand.next_gaussian(),
+        }
+    }
+
+    pub fn next_triangular(&mut self, mode: f64, deviation: f64) -> f64 {
+        mode + deviation * (self.next_f64() - self.next_f64())
+    }
+
+    pub fn skip(&mut self, count: i32) {
+        for _ in 0..count {
+            self.next_i64();
+        }
+    }
+
+    pub fn next_inbetween_i32_exclusive(&mut self, min: i32, max: i32) -> i32 {
+        min + self.next_bounded_i32(max - min)
+    }
+}
+
+pub enum RandomDeriver {
+    Xoroshiro(XoroshiroSplitter),
+    Legacy(LegacySplitter),
+}
+
+impl RandomDeriver {
+    pub fn split_string(&self, seed: &str) -> RandomGenerator {
+        match self {
+            Self::Xoroshiro(deriver) => RandomGenerator::Xoroshiro(deriver.split_string(seed)),
+            Self::Legacy(deriver) => RandomGenerator::Legacy(deriver.split_string(seed)),
+        }
+    }
+
+    pub fn split_u64(&self, seed: u64) -> RandomGenerator {
+        match self {
+            Self::Xoroshiro(deriver) => RandomGenerator::Xoroshiro(deriver.split_u64(seed)),
+            Self::Legacy(deriver) => RandomGenerator::Legacy(deriver.split_u64(seed)),
+        }
+    }
+
+    pub fn split_pos(&self, x: i32, y: i32, z: i32) -> RandomGenerator {
+        match self {
+            Self::Xoroshiro(deriver) => RandomGenerator::Xoroshiro(deriver.split_pos(x, y, z)),
+            Self::Legacy(deriver) => RandomGenerator::Legacy(deriver.split_pos(x, y, z)),
+        }
+    }
+}
```

Comment:
Since these are simple branching functions, I suggest you add the `#[inline(always)]` attribute. to avoid a useleess function call overhead.

Reply:
Shouldn't the compiler do that automatically?

Reply:
It should but it can also decide not to depending on alignement and instructions of the caller.

Reply:
should be good now